### PR TITLE
Move RollingUpdateOnReady to stable

### PR DESCRIPTION
**Pull request by Jeffwan**
**What type of PR is this?**

/kind feature

**What this PR does / Why we need it**:
Move feature gate `RollingUpdateOnReady ` from beta to stable. 

**Which issue(s) this PR fixes**:

Closes https://github.com/googleforgames/agones/issues/2239

**Special notes for your reviewer**:
I check all code references but not sure what doc we want to update. Please help give some guidance on that.

/cc @roberthbailey

### CODE REVIEW

File: pkg/fleets/controller_test.go [hunk 1]
```diff
@@ -1118,11 +1109,6 @@ func TestFeatureRollingUpdateOnReady(t *testing.T) {
 func TestControllerRollingUpdateDeployment(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (Jeffwan) [reply pkg/fleets/controller_test.go:1109]: I will rerun it locally and check the problem

**Comment by roberthbailey** [comment]
We have about a week left before the next release candidate. Do you think you will be able to work on this before the 1.18 release?

**Comment by Jeffwan** [comment]
@roberthbailey Let run try to solve this problem by today. If I can not make it, please take it and help on it.

File: site/content/en/docs/Guides/feature-stages.md [hunk 2]
```diff
@@ -29,7 +29,6 @@ The current set of `alpha` and `beta` feature gates are:
 | Example Gate (not in use) | `Example` | Disabled | None | 0.13.0 |
 | [Player Tracking]({{< ref "/docs/Guides/player-tracking.md" >}}) | `PlayerTracking` | Disabled | `Alpha` | 1.6.0 |
 | [SDK Send GameServer on Watch execution]({{< ref "/docs/Guides/Client SDKs/_index.md#watchgameserverfunctiongameserver" >}}) | `SDKWatchSendOnExecute` | Enabled | `Beta` | 1.12.0 |
-| Fix for RollingUpdate [Scale down](https://github.com/googleforgames/agones/issues/1625) | `RollingUpdateOnReady` | Enabled | `Beta` | 1.14.0 |
```
[the remark this answers is not in the document]
Comment (Jeffwan) [reply site/content/en/docs/Guides/feature-stages.md:32]:
```
{{% feature publishVersion="1.18.0" %}}
| Feature Name | Gate    | Default | Stage | Since |
|--------------|---------|---------|-------|-------|
| Example Gate (not in use) | `Example` | Disabled | None | 0.13.0 |
| [Player Tracking]({{< ref "/docs/Guides/player-tracking.md" >}}) | `PlayerTracking` | Disabled | `Alpha` | 1.6.0 |
| [SDK Send GameServer on Watch execution]({{< ref "/docs/Guides/Client SDKs/_index.md#watchgameserverfunctiongameserver" >}}) | `SDKWatchSendOnExecute` | Enabled | `Beta` | 1.12.0 |
| [Utilize Node ExternalDNS](https://github.com/googleforgames/agones/issues/1921) and additional [details]({{< ref "/docs/FAQ/_index.md" >}}) | `NodeExternalDNS` | Disabled | `Alpha` | 1.12.0 |
| [Custom resync period for FleetAutoscaler](https://github.com/googleforgames/agones/issues/1955) | `CustomFasSyncInterval` | Disabled | `Alpha` | 1.17.0 |
| [GameServer state filtering on GameServerAllocations](https://github.com/googleforgames/agones/issues/1239) | `StateAllocationFilter` | Disabled | `Alpha` | 1.14.0 |
| [GameServer player capacity filtering on GameServerAllocations](https://github.com/googleforgames/agones/issues/1239) | `PlayerAllocationFilter` | Disabled | `Alpha` | 1.14.0 |
{{% /feature %}}

```

Is this the right way?

File: test/e2e/fleet_test.go [hunk 3]
```diff
@@ -297,20 +297,15 @@ func TestFleetRollingUpdate(t *testing.T) {
...
-						if maxUnavailable == 0 {
-							maxUnavailable = 1
-						}
-						// This difference is inevitable, also could be seen with Deployments and ReplicaSets
-						shift = maxUnavailable
+					if maxUnavailable == 0 {
+						maxUnavailable = 1
 					}
+					// This difference is inevitable, also could be seen with Deployments and ReplicaSets
+					shift = maxUnavailable
```
[the remark this answers is not in the document]
Comment (Jeffwan) [reply test/e2e/fleet_test.go:308]: I change it to
```
shift := maxUnavailable
expectedTotal := targetScale + maxSurge + maxUnavailable + shift
``` 

Looks like only 310 has `shift` reference. but the formula already has `maxUnavailable`. I think it's better to keep this meaningful variable.

File: pkg/fleets/controller_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jeffwan) [reply pkg/fleets/controller_test.go:1109]: @roberthbailey I can reproduce the test failures in my env. Seems change in controller.go is straightforward.  I didn't make other changes. Not sure why this test failed.   Do you have any clues?

File: site/content/en/docs/Guides/feature-stages.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (roberthbailey) [reply site/content/en/docs/Guides/feature-stages.md:32]: This will cause the whole table to be hidden until the next release. Unfortunately, with tables you need to copy the whole table, surround the old version in the expiryVersion tag and then surround the new table in the publishVersion tag that you've added here.

**Comment by roberthbailey** [comment]
From the latest run:

```
FAIL
Makefile:259: recipe for target 'test-go' failed
```

I pulled your changes down locally and running `make test-go` is succeeding for me....

File: pkg/fleets/controller_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (markmandel) [reply pkg/fleets/controller_test.go:1109]: Given that these tests were aimed at when the feature was off `assert.NoError(t, utilruntime.ParseFeatures(string(utilruntime.FeatureRollingUpdateOnReady)+"=false"))`, it's expected they would fail now it's enabled always.

So they would either need to be updated to the correct values, or removed entirely.

**Comment by roberthbailey** [comment]
I think I was testing the wrong commits. Let me try again.

File: pkg/fleets/controller_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (roberthbailey) [reply pkg/fleets/controller_test.go:1109]: I took a quick look at these tests and it isn't immediately obvious to me what we are checking for. What test coverage are we losing if we delete the entire set of tests? Or just delete the 3 that are failing?

[the remark this answers is not in the document]
Comment (markmandel) [reply pkg/fleets/controller_test.go:1109]: This tests various gameserver set and rolling update deployment scenarios, and if they go beyond max surge, remove items as appropriate, etc.

Basically does `rollingUpdateDeployment` do as it should.

I think it's worth converting this over before moving to stable - also to ensure an extra layer that it is doing as expected.

[the remark this answers is not in the document]
Comment (roberthbailey) [reply pkg/fleets/controller_test.go:1109]: That makes sense to me. @Jeffwan - we have about 1 more day to try and get this in before the 1.18 release, otherwise we can ship it with 1.19.

[the remark this answers is not in the document]
Comment (Jeffwan) [reply pkg/fleets/controller_test.go:1109]: em. Seem it's better to change to the appropriate values rather than removing them? Let me check algorithm today and see if I can make the changes.

[the remark this answers is not in the document]
Comment (Jeffwan) [reply pkg/fleets/controller_test.go:1109]: I think I figure out the problem. I manually change to use `c.rollingUpdateRestBeforeFixOnReady()` they all pass. They do work for case `utilruntime.FeatureRollingUpdateOnReady=false`. 

https://github.com/googleforgames/agones/blob/aefac73bad27949e745e6354f709c70f1646470b/pkg/fleets/controller.go#L605-L610

Since we already use `rollingUpdateRestFixedOnReady` and remove `c.rollingUpdateRestBeforeFixOnReady`, these test need changes. 


1. I go though the logic and change right number in test cases "test smalled numbers of active and allocated" and "attempt to drive replicas over the max surge" 

2. I remove "statuses don't match the spec. nothing should happen", this is because update won' be called in the past `c.rollingUpdateRestBeforeFixOnReady `. However, this test case is not valid anymore because `c.rollingUpdateRestFixedOnReady ` does consider active GSS.


@roberthbailey @markmandel Please have another check.

**Comment by Jeffwan** [comment]
Is there a way to restart the test? I manually test controller_test.go and it works in my env

**Comment by roberthbailey** [comment]
I've restarted the test.

**Comment by roberthbailey** [comment]
I've created https://github.com/googleforgames/agones/pull/2289 to skip the link check for that site.

**Review by roberthbailey — APPROVED** [review]
